src/main/java/com/labsynch/labseer/domain/StandardizationDryRunCompound.java

```diff
@@ -114,6 +124,99 @@ public StandardizationHistory fetchStats() {
+        EntityManager em = StandardizationDryRunCompound.entityManager();
+        CriteriaBuilder criteriaBuilder = em.getCriteriaBuilder();
+        CriteriaQuery<StandardizationDryRunCompound> criteria = criteriaBuilder.createQuery(StandardizationDryRunCompound.class);
+        Root<StandardizationDryRunCompound> root = criteria.from(StandardizationDryRunCompound.class);
+        criteria.select(root);
+
+		// Predicate List
+		List<Predicate> predicates = new ArrayList<Predicate>();
+
+		// New mol weight
+		if (dryRunSearch.getNewMolWeight() != null) {
+			predicates = buildPredicateFromNumericValue(criteriaBuilder, root, predicates, "newMolWeight", dryRunSearch.getNewMolWeight().getValue(), dryRunSearch.getNewMolWeight().getOperator());
+		}
+
+		// Delta mol weight
+		if (dryRunSearch.getDeltaMolWeight() != null) {
+			predicates = buildPredicateFromNumericValue(criteriaBuilder, root, predicates, "deltaMolWeight", dryRunSearch.getDeltaMolWeight().getValue(), dryRunSearch.getDeltaMolWeight().getOperator());
+		}
+
+		// Old mol weight
+		if (dryRunSearch.getOldMolWeight() != null) {
+			predicates = buildPredicateFromNumericValue(criteriaBuilder, root, predicates, "oldMolWeight", dryRunSearch.getOldMolWeight().getValue(), dryRunSearch.getOldMolWeight().getOperator());
+		}
+
+		// Corp name in list
+		if (dryRunSearch.getCorpNames() != null && dryRunSearch.getCorpNames().length > 0) {
+			predicates.add(root.get("corpName").in(dryRunSearch.getCorpNames()));
+		}
+
+		// Existing duplicate
+		if (dryRunSearch.getHasExistingDuplicates() != null) {
+			if(dryRunSearch.getHasExistingDuplicates()) {
+				predicates.add(root.get("existingDuplicates").isNotNull());
+			} else {
+				predicates.add(root.get("existingDuplicates").isNull());
+			}
+		}
+
+		// New duplicates
+		if (dryRunSearch.getHasNewDuplicates() != null) {
+			if(dryRunSearch.getHasNewDuplicates()) {
+				predicates.add(root.get("newDuplicates").isNotNull());
+			} else {
+				predicates.add(root.get("newDuplicates").isNull());
+			}
+
+		}
+
+		// Boolean searches
+		// Changed structure
+		if (dryRunSearch.getChangedStructure() != null) {
+			predicates.add(criteriaBuilder.equal(root.get("changedStructure"), dryRunSearch.getChangedStructure()));
+		}
+
+		// Display change
+		if (dryRunSearch.getDisplayChange() != null) {
+			predicates.add(criteriaBuilder.equal(root.get("displayChange"), dryRunSearch.getDisplayChange()));
+		}
+
+		// As drawn display change
+		if (dryRunSearch.getAsDrawnDisplayChange() != null) {
+			predicates.add(criteriaBuilder.equal(root.get("asDrawnDisplayChange"), dryRunSearch.getAsDrawnDisplayChange()));
+		}
+
+		if(predicates.size() > 0) {
+			Predicate[] predicatesToAdd = new Predicate[0];
+			predicatesToAdd = predicates.toArray(predicatesToAdd);
+			Predicate firstItxAndPredicates = criteriaBuilder.and(predicatesToAdd);
+			criteria.where(predicates.toArray(new Predicate[predicates.size()]));
```

Comment:
Something looks off here. You're not using that `firstItxAndPredicates` variable you just created. Shouldn't this line just be `criteria.where(firstItxAndPredicates)` or something?